### PR TITLE
Adding chest-xray-covid model

### PR DESCRIPTION
**Pull request by Rishit-dagli**
The COVID Chest X-Ray model is a lightweight Deep Learning model for image classification where the goal is to predict from a Chest X-Ray if a person might have COVID. Currently, only the TF Lite implementation of the model is added.

### CODE REVIEW

**Comment by maringeo** [comment]
Hi @Rishit-dagli,

Since this is your first time contributing to the TFhub.dev repository, we just need you to read & agree to the TensorFlow Hub terms of service at https://tfhub.dev/terms. If you feel comfortable moving forward, please add a comment "I have read and agree to the TensorFlow Hub Terms of Service" to this thread.

**Comment by Rishit-dagli** [comment]
Hi @maringeo ,

I have read and agree to the TensorFlow Hub Terms of Service

File: tfhub_dev/assets/rishit-dagli/rishit-dagli.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Rishit Dagli.
+
+[![Icon URL]](https://avatars3.githubusercontent.com/u/39672672?s=460&u=315923c86d9e20cca52085af7eb3190c5bb263e7&v=4)
+
+## Details
+Rishit Dagli ([@rishit_dagli](https://twitter.com/rishit_dagli)).
+10 th Grade Student.
+TedX, TedEd Speaker.
+
+**GitHub:** [Rishit-dagli](https://github.com/sayakpaul)\
```
Comment (maringeo) [inline tfhub_dev/assets/rishit-dagli/rishit-dagli.md:11]: The link is probably incorrect. Should it be https://github.com/Rishit-dagli ?

**Comment by Rishit-dagli** [comment]
A query - Since, the validator does not support `format: lite` is there anything else I would need to do to specify so.

I also had another query about publishing for this same model which was not covered clearly in the publishing documentation. Let's say I also want to add TF 2.0 Saved Model as I understand I would have a model directory like this?

```
chest-xray-covid
├── default/lite
 |       └──  1.md
├── saved_model_2
 |      └──  1.md
├── 1.md
 ```

File: tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Placeholder rishit-dagli/chest-xray-covid/1
+Lightweight deep learning model for clasifying chest xrays as COVID or normal.
+
+<!-- module-type: image-classification -->
+<!-- network-architecture: other -->
+<!-- dataset: other -->
+<!-- fine-tunable: false -->
+<!-- language: en -->
```
Comment (maringeo) [inline tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md:8]: The language metadata field is usually used for text models, for example to specify the language of the dataset the model was trained on. Since this is an image model we can remove this field - let me know if I'm missing anything.

File: tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Placeholder rishit-dagli/chest-xray-covid/1
+Lightweight deep learning model for clasifying chest xrays as COVID or normal.
+
+<!-- module-type: image-classification -->
+<!-- network-architecture: other -->
```
Comment (maringeo) [inline tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md:5]: Is it possible to have a more specific description of the network-architecture? Same for dataset.

**Review by maringeo — CHANGES_REQUESTED** [review]

File: tfhub_dev/assets/rishit-dagli/rishit-dagli.md (hunk 1 above)
Comment (Rishit-dagli) [reply tfhub_dev/assets/rishit-dagli/rishit-dagli.md:11]: This was my first time contributing so as a reference to the directory structure I took inspiration from @sayakpaul models, seems like I forgot to edit this part. Thanks for pointing it out.

**Comment by maringeo** [comment]
> A query - Since, the validator does not support `format: lite` is there anything else I would need to do to specify so.
> 
> I also had another query about publishing for this same model which was not covered clearly in the publishing documentation. Let's say I also want to add TF 2.0 Saved Model as I understand I would have a model directory like this?
> 
> ```
> chest-xray-covid
> ├── default/lite
> |   └──  1.md
> ├── saved_model_2
> |   └──  1.md
> └──  1.md
> ```

`format` is a confusing name. It can be either `<!-- format: saved_model_2 -->` if exported from TF2 SavedModel or `<!-- format: hub -->` if exported from TF1.

If you want to add a TF 2.0 SavedModel, you can delete the current Placeholder MD and set the parent-model of the TF Lite MD to be the TF2 SavedModel MD.

That would result in the following directory structure:
chest-xray-covid
-- default/lite
-- --1.md
-- 1.md (this is the SavedModel MD)

File: tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md (hunk 3 above)
Comment (Rishit-dagli) [reply tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md:5]: I use a Neural Architecture Search based method to build the model and thus was not sure if I should fill out the architecture.

File: tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md (hunk 2 above)
Comment (Rishit-dagli) [reply tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md]: Thanks for pointing that out, I have now removed the field

File: tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md (hunk 3 above)
Comment (Rishit-dagli) [reply tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md:5]: I have added the dataset field.

**Comment by Rishit-dagli** [comment]
> > A query - Since, the validator does not support `format: lite` is there anything else I would need to do to specify so.
> > I also had another query about publishing for this same model which was not covered clearly in the publishing documentation. Let's say I also want to add TF 2.0 Saved Model as I understand I would have a model directory like this?
> > ```
> > chest-xray-covid
> > ├── default/lite
> > |   └──  1.md
> > ├── saved_model_2
> > |   └──  1.md
> > └──  1.md
> > ```
> 
> `format` is a confusing name. It can be either `<!-- format: saved_model_2 -->` if exported from TF2 SavedModel or `<!-- format: hub -->` if exported from TF1.
> 
> If you want to add a TF 2.0 SavedModel, you can delete the current Placeholder MD and set the parent-model of the TF Lite MD to be the TF2 SavedModel MD.
> 
> That would result in the following directory structure:
> chest-xray-covid
> -- default/lite
> -- --1.md
> -- 1.md (this is the SavedModel MD)

Thank you, this solves it but a quick follow up question on this let's say I want to add two more model types, how would I go about that, would I use something like this?

```
-- default/lite
-- --1.md
-- coral (let's say)
-- --1.md
-- 1.md (this is the SavedModel MD)
```

**Comment by maringeo** [comment]
> Thank you, this solves it but a quick follow up question on this let's say I want to add two more model types, how would I go about that, would I use something like this?
> 
> ```
> -- default/lite
> -- --1.md
> -- coral (let's say)
> -- --1.md
> -- 1.md (this is the SavedModel MD)
> ```

The typical structure is
--default/lite
-- -- 1.md (lite MD)
-- default/coral
-- -- 1.md (coral MD)
-- default/tfjs
-- -- 1.md (tfjs MD)
-- 1.md (SavedModel MD, or Placeholder MD if there is no SavedModel MD)

We're working on documenting this better (and also simplifying the structure). Sorry for the inconvenience.

**Comment by Rishit-dagli** [comment]
> > Thank you, this solves it but a quick follow up question on this let's say I want to add two more model types, how would I go about that, would I use something like this?
> > ```
> > -- default/lite
> > -- --1.md
> > -- coral (let's say)
> > -- --1.md
> > -- 1.md (this is the SavedModel MD)
> > ```
> 
> The typical structure is
> --default/lite
> -- -- 1.md (lite MD)
> -- default/coral
> -- -- 1.md (coral MD)
> -- default/tfjs
> -- -- 1.md (tfjs MD)
> -- 1.md (SavedModel MD, or Placeholder MD if there is no SavedModel MD)
> 
> We're working on documenting this better (and also simplifying the structure). Sorry for the inconvenience.

Thank you, this solves the query.

File: tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Placeholder rishit-dagli/chest-xray-covid/1
+Lightweight deep learning model for clasifying chest xrays as COVID or normal.
```
Comment (maringeo) [inline tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md:2]: I assume this is a binary classifier, could we rephrase this as `COVID or not` instead of `COVID or normal`. Same for the rest of the MD files.

Comment (Rishit-dagli) [reply tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md:2]: Sure! That seems to be a better way of writing this.

File: tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md (hunk 3 above)
Comment (Rishit-dagli) [reply tfhub_dev/assets/rishit-dagli/chest-xray-covid/1.md:5]: @maringeo Does this look good or would I need to enter the architecture details?

**Comment by maringeo** [comment]
Hi @Rishit-dagli , sorry I missed this in the previous review iteration - is it possible to extend the Lite MD with a brief description of the model performance (for example ROC, precision-recall or another performance metric that you've used when training/evaluating the model)?

**Comment by Rishit-dagli** [comment]
> Hi @Rishit-dagli , sorry I missed this in the previous review iteration - is it possible to extend the Lite MD with a brief description of the model performance (for example ROC, precision-recall or another performance metric that you've used when training/evaluating the model)?

@maringeo  This would definitely be helpful to others trying to use this model, Thanks for pointing that out. I have added the model [Precision/ Recall curve](https://storage.googleapis.com/trained-covid-model/lite_metrics/pr_curve.PNG) and the [Confusion matrix](https://storage.googleapis.com/trained-covid-model/lite_metrics/confusion_matrix.PNG) for trained Lite model.

**Comment by maringeo** [comment]
Hi @Rishit-dagli , thank you for contributing to TF Hub. 

The model is live on https://tfhub.dev/rishit-dagli/lite-model/chest-xray-covid/1/default/1. 

This PR has been merged in 47972a85945f356da62e1dea732b7ddabf937e03.

**Comment by Rishit-dagli** [comment]
@maringeo I just noticed TF Hub also rendering the model metadata on the page, looks really cool!

**Comment by sayakpaul** [comment]
@Rishit-dagli congratulations on this one. Some thoughts:

- There's a minor typo I noticed - "Lightweight deep learning model for **clasifying** chest xrays as COVID or not." 
- If you have a demo app already it's probably better to have that linked from your TF-Hub model page. 

@maringeo I see that the dataset property is missing from the model card. Is this issue known?

**Comment by Rishit-dagli** [comment]
> @Rishit-dagli congratulations on this one. Some thoughts:
> 
> * There's a minor typo I noticed - "Lightweight deep learning model for **clasifying** chest xrays as COVID or not."
> * If you have a demo app already it's probably better to have that linked from your TF-Hub model page.
> 
> @maringeo I see that the dataset property is missing from the model card. Is this issue known?

Thanks, @sayakpaul for pointing out these errors

- I will update this PR with the typo being fixed
- I have a demo app but I would need to work on the UI part a bit to make it usable so I will open a PR in the future with these changes

About the dataset property, the TF Hub metadata mentions the dataset name as `chest-xray-covid` and I can see the model listed [here](https://tfhub.dev/s?dataset=chest-xray-covid) from the dataset search but I can't see the dataset poperty on the model page, maybe that needs to be fixed.

**Comment by sayakpaul** [comment]
Enlisting a not-fully-worked out APP (i.e. not so cool UI) usually works as long as it demonstrates the model usage in a systematic manner.

**Comment by kempy** [comment]
Glad to hear you like the TFlite metadata preview!

> I have seen that TF Hub also supports having an Open in Android Studio Button to use with the model binding plugin, since the docs do not mention anything about that could you also share details about that.

This just means you can download this model and import it into Android Studio with ML Model Binding. Learn more at https://developer.android.com/studio/preview/features#tensor-flow-lite-models

> I see that the dataset property is missing from the model card. Is this issue known?

Yep, known issue, we'll try to fix it for this model this week and get it fixed generally in the next few weeks since it happens frequently for new datasets

**Comment by sayakpaul** [comment]
Cool @kempy. Looking forward to those updates. Metadata preview is definitely a game changer given the usefulness it offers.

**Comment by Rishit-dagli** [comment]
> Glad to hear you like the TFlite metadata preview!
> 
> > I have seen that TF Hub also supports having an Open in Android Studio Button to use with the model binding plugin, since the docs do not mention anything about that could you also share details about that.
> 
> This just means you can download this model and import it into Android Studio with ML Model Binding. Learn more at https://developer.android.com/studio/preview/features#tensor-flow-lite-models
> 
> > I see that the dataset property is missing from the model card. Is this issue known?
> 
> Yep, known issue, we'll try to fix it for this model this week and get it fixed generally in the next few weeks since it happens frequently for new datasets

@kempy Thank you! I had made an Android project with this model (a basic one) I will very soon add it.
